### PR TITLE
Allow 'playsinline' player option

### PR DESCRIPTION
**Pull request by alex-barstow**
### Description
Video.js players can accept a number of standard `<video>` element options (`autoplay`, `muted`, `loop`, etc), but not currently `playsinline`, which is now part of the [HTML spec](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-video-playsinline). We should add it to the list of `<video>` attributes that can be provided to the player as options.

### Specific Changes proposed
Update the `Html5` and `Player` classes with getter and setter methods for `playsinline` and make a few other additions to ensure that the `playsinline` option works the same way as the other `<video>` element options.

### Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/axedog/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by alex-barstow** [comment]
I assume this will need a sister PR in `master` as well

File: src/js/player.js [hunk 1]
```diff
@@ -2381,6 +2382,26 @@ class Player extends Component {
   }
 
   /**
+   * Get or set the playsinline attribute.
+   *
+   * @param {boolean} [value]
+   *        - true means that we should play inline in iOS Safari
```
Comment (gkatsev) [inline src/js/player.js]: I would change these two say `set the playsinline attribute` or `unset it`. And mention above that this probably only applies on ios right now.

File: src/js/tech/html5.js [hunk 2]
```diff
@@ -1235,6 +1235,20 @@ Html5.resetMediaElement = function(el) {
   'autoplay',
 
   /**
+   * Get the value of `playsinline` from the media element. `playsinline` indicates
+   * that the media should play inline in iOS Safari.
```
Comment (gkatsev) [inline src/js/tech/html5.js]: iOS is an example of a browser that will only play inline if the attribute is present.

File: src/js/tech/html5.js [hunk 3]
```diff
@@ -1518,6 +1532,19 @@ Html5.resetMediaElement = function(el) {
   'autoplay',
 
   /**
+   * Set the value of `playsinline` on the media element. `playsinline` indicates
```
Comment (gkatsev) [inline src/js/tech/html5.js]: Same here.

**Comment by gkatsev** [comment]
Also, yes, a PR against master would be good. Can be done once this PR is approved, though.

**Comment by gkatsev** [comment]
@alex-barstow While testing, I found that if I have a video embed with `playsinline` attribute, calling `player.playsinline()` after the player loads doesn't reflect that value.

Also, we should add an `@abstract` method in `tech.js` for this.

File: src/js/tech/html5.js [hunk 4]
```diff
@@ -1431,6 +1431,27 @@ Html5.resetMediaElement = function(el) {
...
+ *
+ * @method Html5#playsinline
+ * @return {boolean}
+ *         - The value of `playsinline` from the media element.
+ *         - True indicates that the media should play inline.
+ *         - False indicates that the media should not play inline.
+ *
+ * @see [Spec]{@link https://html.spec.whatwg.org/#attr-video-playsinline}
+ */
+Html5.prototype.playsinline = function() {
```
Comment (gkatsev) [inline src/js/tech/html5.js]: you should move these up and use the class syntax

File: src/js/tech/html5.js [hunk 5]
```diff
@@ -1553,6 +1574,30 @@ Html5.resetMediaElement = function(el) {
   };
 });
 
+// This setter is declared outside of the loop because there is no 'playsinline'
```
Comment (gkatsev) [inline src/js/tech/html5.js]: We can just describe this in the PR instead, don't think it's necessary, especially if we move it into the class syntax.

File: src/js/player.js [hunk 6]
```diff
@@ -2381,6 +2382,29 @@ class Player extends Component {
   }
 
   /**
+   * Set or unset the playsinline attribute.
+   * Playsinline tells the browser that non-fullscreen playback is preferred.
+   *
```
Comment (gkatsev) [inline src/js/player.js]: Can we add a `@see` here as well?

**Comment by imbcmdth** [comment]
@gkatsev @alex-barstow  Should we have any testing like https://github.com/videojs/video.js/blob/master/test/unit/player.test.js#L94 where we make sure `autoplay` is propagated from the video element?

**Comment by gkatsev** [comment]
@imbcmdth Perhaps adding it would be useful, but the tests are already testing that if attributes are present they persist when videojs wraps the video element.
I guess we might want to test the setter as well.

**Comment by gkatsev** [comment]
@alex-barstow would you be able to make the sister PR against master for this as well?

**Pull request merged**